scipy/signal/_ltisys.py

```diff
@@ -2356,7 +2370,12 @@ def impulse2(system, X0=None, T=None, N=None, **kwargs):
     # Move the impulse in the input to the initial conditions, and then
     # solve using lsim2().
     ic = B + X0
-    Tr, Yr, Xr = lsim2(sys, T=T, X0=ic, **kwargs)
+    with warnings.catch_warnings():
+        warnings.filterwarnings(action='ignore', message="lsim2 is deprecated " 
+            "and will be removed from scipy. "
+            "Use the feature equivalent lsim function.", 
+            category=DeprecationWarning)
+        Tr, Yr, Xr = lsim2(sys, T=T, X0=ic, **kwargs)
```

Comment:
Can we not translate this already to using `lsim` directly? Or do we just feel there's no point because `impulse2` itself will be removed as well?

Reply:
`**kwargs` for `lsim2` are not supported in `lsim`, so I cannot pass them through. I could drop them silently, but this could lead to backward incompabilities I assume. 
I thought this is fine, as `impulse2` will be removed as well

Reply:
Yeah, with removing `impulse2` I agree that this is acceptable.